java/maven/src/org/netbeans/modules/maven/NbMavenProjectImpl.java

```diff
@@ -639,6 +640,24 @@ public MavenEmbedder getEmbedder() {
         return uris.toArray(new URI[uris.size()]);
     }
 
+    private static File findJavaRoot(File root, boolean test) {
+        String javaFolder = test ? "test" + File.separator + "java" : "main" + File.separator + "java"; // NOI18N
+        File newRoot = new File(root, javaFolder);
+        if (newRoot.isDirectory()) {
+            return newRoot;
+        }
+        File[] listFiles = root.listFiles();
+        if (listFiles != null) {
+            for (File f : listFiles) {
+                newRoot = new File(f, javaFolder);
+                if (newRoot.isDirectory()) {
+                    return newRoot;
+                }
+            }
+        }
```

Comment:
This will break, if an existing plugin (maven-compiler (annotations), jaxb, ...) generates sources in the package `test.java` or `main.java`. I don't think, that that is impossible. You could analyze java files there for the package declaration, but that sounds like overkill.

IMHO it would be better if the plugin would be fixed to use the conventional directory: `target/generated-sources/<PLUGIN>`. At least `maven-compiler-plugin`  and `jaxb2-maven-plugin` both use that and the existing code in the netbeans code is an indicator, that the original authors came to the same conclusion.